code/datums/progressbar.dm

```diff
@@ -134,6 +134,8 @@
 		return
 	last_progress = progress
 	bar.icon_state = "prog_bar_[round(((progress / goal) * 100), 5)]"
+	if(progress == goal)
+		end_progress()
```

Comment:
This isn't correct, the progress bar works in other instances, the termination of the progress has to be done caller side, aka you need to handle the progress completion from the caller (the borer code, most likely)